galpy/potential/TwoPowerSphericalPotential.py

```diff
@@ -93,25 +129,21 @@ def _evaluate(self,R,z,phi=0.,t=0.,_forceFloatEval=False):
         HISTORY:
            2010-07-09 - Started - Bovy (NYU)
         """
-        if not _forceFloatEval and not self.integerSelf == None:
+        if not _forceFloatEval and self.integerSelf is not None:
             return self.integerSelf._evaluate(R,z,phi=phi,t=t)
         elif self.beta == 3.:
-            r= numpy.sqrt(R**2.+z**2.)
-            return (1./self.a)\
-                *(r-self.a*(r/self.a)**(3.-self.alpha)/(3.-self.alpha)\
-                      *special.hyp2f1(3.-self.alpha,
-                                      2.-self.alpha,
-                                      4.-self.alpha,
-                                      -r/self.a))/(self.alpha-2.)/r
-        else:
-            r= numpy.sqrt(R**2.+z**2.)
-            return special.gamma(self.beta-3.)\
-                *((r/self.a)**(3.-self.beta)/special.gamma(self.beta-1.)\
-                      *special.hyp2f1(self.beta-3.,
-                                      self.beta-self.alpha,
-                                      self.beta-1.,
-                                      -self.a/r)
-                  -special.gamma(3.-self.alpha)/special.gamma(self.beta-self.alpha))/r
+            a, alpha, beta = self.a, self.alpha, self.beta
+            r= sqrt(R**2.+z**2.)
+            return (1./a)\
+                *(r-a*(r/a)**(3.-alpha)/(3.-alpha)\
+                      *hyp2f1(3.-alpha, 2.-alpha, 4.-alpha, -r/a))/(alpha-2.)/r
+        # else:
+        a, alpha, beta = self.a, self._alpha, self._beta
+        r= sqrt(R**2.+z**2.)
+        return Gamma(beta-3.)\
+            *((r/a)**(3.-beta)/Gamma(beta-1.)\
+                  *hyp2f1(beta-3., beta-alpha, beta-1., -a/r)
+              -Gamma(3.-alpha)/Gamma(beta-alpha))/r
```

Comment:
From a glance through this, nothing seems to have changed; why did you re-write this?

Reply:
for the SciPy special function spaces.